doc/nrf/config_and_build/bootloaders_and_dfu/bootloader.rst

```diff
@@ -106,18 +108,46 @@ If the verification fails, the boot process stops.
 This way, the immutable bootloader can guarantee that the next image in the boot sequence will not start up if it has been tampered with in any way.
 For example, if an attacker attempts to take over the device by altering the firmware, the device will not boot, and thus not run the infected code.
 
-The immutable bootloader is locked to the flash memory and cannot be modified or deleted without erasing the entire device.
-For the |NSIB|, see :ref:`Locking the flash memory <bootloader_rot>` for more information.
+More specifically, the immutable bootloader performs the following steps when it runs:
 
-Except for providing your own keys, there is no need to modify the immutable bootloader in any way before you program it.
-The verification provided by this bootloader is recommended and suitable for all the most common user scenarios and includes the following checks:
+1. Locking of the flash memory.
 
-* Signature verification - Verifies that the key used for signing the next image in the boot sequence matches one of the provided public keys.
-* Metadata verification - Checks that the images are compatible.
+   To enable the RoT, the immutable bootloader locks the flash memory address range containing itself and its configuration using the hardware available on the given architecture.
+   (The immutable bootloader cannot be modified or deleted without erasing the entire device.)
 
-.. caution::
-   You must generate and use your own signing keys while in development and before deploying when using either MCUboot or the |NSIB| as an immutable bootloader.
-   See :ref:`ug_fw_update_development_keys` for more information.
+#. Selection of the next slot in the boot chain.
+
+   The next stage in the boot chain can either be an application or another bootloader:
+
+   * When the next stage is an application, you can allocate one or two slots for it.
+   * When the next stage is a second-stage bootloader, two slots are always used.
+
+   Each slot has a version number, and the bootloader will select the slot with the latest version.
+   For more information about creating a second stage bootloader, see :ref:`ug_bootloader_adding_upgradable`.
+
+#. Verification of the next stage in the boot chain.
+
+   The verification provided by this bootloader is recommended and suitable for all the most common user scenarios and includes the following checks:
+
+   * Signature verification - Verifies that the key used for signing the next image in the boot sequence matches one of the provided public keys.
+
+     The next image has metadata containing the full public key that corresponds to the private key used to sign the firmware.
+     This public key is checked against the provisioned hashes of public keys to determine if the image is valid.
```

Comment:
It would be nice to be a bit more specific than "image is valid" I think.
Maybe we can say that it both checks that the image is from only you (authenticity) and it also checks that the image did not change by accident (integrity).

Reply:
Rephrased.

Reply:
Rephrasing looks good to me